src/pages/vans/Vans.jsx

```diff
@@ -0,0 +1,121 @@
+import { useState } from "react";
+import { Link, useSearchParams, useLoaderData } from "react-router-dom";
+import { getVans } from "../../api";
+
+export function loader() {
+  return getVans();
+}
+
+const Vans = () => {
+  const info = useLoaderData();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [error, setError] = useState(null);
+  const typeFilter = searchParams.get("type");
+
+  const filtered = typeFilter
+    ? info.filter((char) => char.type === typeFilter)
+    : info;
+
+  const handleFilterChange = (key, value) => {
+    setSearchParams((prevParams) => {
+      if (value === null) {
+        prevParams.delete(key);
+      } else {
+        prevParams.set(key, value);
+      }
+      return prevParams;
+    });
+  };
+
+  if (error) {
+    return (
+      <h1 className="text-2xl font-black text-[#161616] bg-[#FFF7ED] leading-6">
+        There was an error: {error.message}
+      </h1>
+    );
+  }
+
```

Comment:
I don't think you still need to throw this error message in this file since you have an error handling component that are wrapping all routes.